src/building/bootstrapping.md

```diff
@@ -47,54 +62,59 @@ graph TD
     classDef with-s1c fill: lightgreen;
 ```
 
-### Stage 0
+[objects]: https://en.wikipedia.org/wiki/Object_code
+
+### The stages: how each compiler is obtained
+
+#### Stage 0: the pre-compiled compiler
 
-The stage0 compiler is usually the current _beta_ `rustc` compiler
-and its associated dynamic libraries,
-which `x.py` will download for you.
-(You can also configure `x.py` to use something else.)
+A pre-compiled compiler and its set of dependencies are downloaded. By default,
+it is the current beta release. This is the stage 0 compiler.
 
-The stage0 compiler is then used only to compile `rustbuild`, `std`, and `rustc`.
-When compiling `rustc`, the stage0 compiler uses the freshly compiled `std`.
-There are two concepts at play here:
-a compiler (with its set of dependencies)
-and its 'target' or 'object' libraries (`std` and `rustc`).
-Both are staged, but in a staggered manner.
+#### Stage 1: from current code, by an earlier compiler
 
-### Stage 1
+The stage 0 compiler compiles from current code `rustbuild` and `std` and uses
+them to compile from current code a compiler. This is the stage 1 compiler.
```

Comment:
*produce* (or an equivalent) is an important word to keep here... makes it more clear what is happenin

Reply:
Resolve in favor of https://github.com/rust-lang/rustc-dev-guide/pull/1458#discussion_r1048000762 ?